### PR TITLE
action-matcher/mvp

### PR DESCRIPTION
**Pull request by ksingarajah**
It's working but it's incredibly ugly... Executor to action matching connection is working, but errors arise when calls are made to BotActions. Still need to implement: GET_SETTING, GET_STAT, GET_SETTINGS, GET_STATS, NAVIGATE_TO_CONVERSATION, NAVIGATE, SET_SETTING. Other actions have been tested with a few test commands. Also need to modularise the matchAction method and implement unit testing.

### CODE REVIEW

File: src/main/java/codeu/bot/ActionMatcher.java [hunk 1]
```diff
@@ -0,0 +1,372 @@
...
+        ArrayList<Integer> spaceIndices = new ArrayList<Integer>();
+
+        int findIndex = input.indexOf(" ", 0);
+        while (findIndex >= 0) {
+            spaceIndices.add(findIndex);
+            findIndex = input.indexOf(" ", findIndex+1);
+        }
+
+        outerloop:
+        for (String element : set) {
```
Comment (ricglz) [inline src/main/java/codeu/bot/ActionMatcher.java:83]: What exactly is set ?

**Review by ricglz — APPROVED** [review]
Overall I think that is good code. It certainly needs a little bit more of love and care, but that can be solved. Also I think that it would be nice to check how to improve the complexity, to made it more efficient.

File: src/main/java/codeu/bot/ActionMatcher.java (hunk 1 above)
Comment (manjilgtm) [reply src/main/java/codeu/bot/ActionMatcher.java:83]: +1

**Review by manjilgtm — APPROVED** [review]
Nice, Implementation of NLP library.

**Pull request merged**